projects/igniteui-angular/src/lib/directives/button/button.directive.ts

```diff
@@ -128,7 +128,8 @@ export class IgxButtonDirective extends IgxButtonBaseDirective implements AfterC
     }
 
     public ngAfterContentInit() {
-        this.nativeElement.addEventListener('click', this.emitSelected.bind(this));
+        this.emitSelected = this.emitSelected.bind(this);
+        this.nativeElement.addEventListener('click', this.emitSelected);
```

Comment:
I believe this can be done with `HostBinding`?

Reply:
You're right. Done.